setup/base/base.go

```diff
@@ -146,6 +148,15 @@ func NewBaseDendrite(cfg *config.Dendrite, componentName string, options ...Base
 		logrus.WithError(err).Panicf("failed to start opentracing")
 	}
 
+	var fts *fulltext.Search
+	isSyncOrMonolith := componentName == "syncapi" || isMonolith
```

Comment:
This just makes me think we should probably define some consts for the various component names somewhere, so that we don't accidentally change them in one place and not another later on. Happy for that to be in a separate PR.